Core/WebCacheManager.swift

```diff
@@ -101,62 +103,61 @@ public class WebCacheManager {
                 _ = group.wait(timeout: .now() + 5)
                 DispatchQueue.main.async {
                     completion()
                 }
             }
         }
-        
-    }
 
-    /// The Fire Button does not delete the user's DuckDuckGo search settings, which are saved as cookies. Removing these cookies would reset them and have undesired
-    ///  consequences, i.e. changing the theme, default language, etc.  These cookies are not stored in a personally identifiable way. For example, the large size setting
-    ///  is stored as 's=l.' More info in https://duckduckgo.com/privacy
-    private func isDuckDuckGoOrAllowedDomain(cookie: HTTPCookie, domain: String) -> Bool {
-        return cookie.domain == domain || (cookie.domain.hasPrefix(".") && domain.hasSuffix(cookie.domain))
     }
 
     public func clear(dataStore: WebCacheManagerDataStore = WKWebsiteDataStore.default(),
-                      appCookieStorage: CookieStorage = CookieStorage(),
                       logins: PreserveLogins = PreserveLogins.shared,
                       completion: @escaping () -> Void) {
-        extractAllowedCookies(from: dataStore.cookieStore, cookieStorage: appCookieStorage, logins: logins) {
-            self.clearAllData(dataStore: dataStore, completion: completion)
-        }
-    }
 
-    private func clearAllData(dataStore: WebCacheManagerDataStore, completion: @escaping () -> Void) {
-        dataStore.removeAllData(completion: completion)
-    }
-    
-    private func extractAllowedCookies(from cookieStore: WebCacheManagerCookieStore?,
-                                       cookieStorage: CookieStorage,
-                                       logins: PreserveLogins,
-                                       completion: @escaping () -> Void) {
-        
-        guard let cookieStore = cookieStore else {
-            completion()
-            return
-        }
-        
-        cookieStore.getAllCookies { cookies in
-            for cookie in cookies {
-                if cookie.domain == Constants.cookieDomain || logins.isAllowed(cookieDomain: cookie.domain) {
-                    cookieStorage.setCookie(cookie)
+        dataStore.removeAllDataExceptCookies {
+            guard let cookieStore = dataStore.cookieStore else {
+                completion()
+                return
+            }
+
+            cookieStore.getAllCookies { cookies in
+                let group = DispatchGroup()
+                let cookiesToRemove = cookies.filter { !logins.isAllowed(cookieDomain: $0.domain) && $0.domain != Constants.cookieDomain }
+
+                for cookie in cookiesToRemove {
+                    group.enter()
+                    cookieStore.delete(cookie) {
+                        group.leave()
+                    }
+                }
+
+                DispatchQueue.global(qos: .background).async {
+                    _ = group.wait(timeout: .now() + 5)
+                    Pixel.fire(pixel: .cookieDeletionTimedOut)
```

Comment:
But aren't we supposed to fire that when there is a timeout only? :)

Reply:
D'oh, yes sorry, am not able to focus properly right now. 🙂 I'll change this to check against the timeout result.

Reply:
Done